### PR TITLE
Add handle to modal-actions when scroll reach the end of modal-content

### PR DESCRIPTION
**Pull request by Klynger**
#### What problem is this solving?

As the title says

#### How should this be manually tested?

1. Open the [workspace](https://quickview--storecomponents.myvtex.com/shoes?_q=shoes&map=ft)
2. Hover the `product-summary`
3. Click on the expand icon that will appear at the top-right corner of the product summary
4. Scroll the content, you should see the style of the `modal-actions` changing.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [X] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
X | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/ModalContent.tsx [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
+        intersectionRef.current.observe(sentinelRef.current)
+      }
+    }
+
+    return () => {
+      if (intersectionRef.current) {
+        intersectionRef.current.disconnect()
+      }
+    }
+  }, [handleIntersection])
```
Comment (klzns) [inline react/ModalContent.tsx]: Can we move this code to a separate module?

File: react/ModalHeader.tsx [hunk 2]
```diff
@@ -57,10 +57,10 @@ export default function ModalHeader(props: Props) {
             onClick={handleClose}
             className={`${handles.closeButton} ma0 bg-transparent pointer bw0 pa3`}
           >
-            <IconClose size={24} type="line" />
+            <IconClose size={32} type="line" />
```
Comment (victorhmp) [inline react/ModalHeader.tsx]: I would either create an `iconCloseSize` prop or use the `icon-close` block here, I can see people wanting to change this `size`

File: react/modules/useIntersection.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+export default function useIntersection(params: Params) {
+  const { dispatch, intersectionRef, sentinelRef } = params
+  const handleIntersection: IntersectionObserverCallback = useCallback(
+    entries => {
+      if (entries[0]) {
+        dispatch({
+          type: 'SET_END_OF_CONTENT',
+          payload: { endOfContent: entries[0].isIntersecting },
+        })
```
Comment (klzns) [inline react/modules/useIntersection.ts]:
```suggestion
        callback(entries[0].isIntersecting)
```

File: react/modules/useIntersection.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { Dispatch } from '../components/ModalContext'
+
+interface Params {
+  dispatch: Dispatch
+  intersectionRef: React.MutableRefObject<IntersectionObserver | null>
+  sentinelRef: React.RefObject<HTMLDivElement | null>
+}
+
+export default function useIntersection(params: Params) {
+  const { dispatch, intersectionRef, sentinelRef } = params
```
Comment (klzns) [inline react/modules/useIntersection.ts]:
```suggestion
  const { callback, intersectionRef, sentinelRef } = params
```

File: react/ModalContent.tsx [hunk 5]
```diff
@@ -1,22 +1,37 @@
...
 
 const CSS_HANDLES = ['contentContainer']
 
 const ModalContent: React.FC = props => {
   const { children } = props
+  const intersectionRef = useRef<IntersectionObserver | null>(null)
+  const sentinelRef = useRef<HTMLDivElement | null>(null)
   const handles = useCssHandles(CSS_HANDLES)
+  const dispatch = useModalDispatch()
+  useIntersection({ intersectionRef, dispatch, sentinelRef })
```
Comment (klzns) [inline react/ModalContent.tsx]:
```suggestion
  useIntersection({
    intersectionRef,
    callback: (isIntersecting) => dispatch({
      type: 'SET_END_OF_CONTENT',
      payload: { endOfContent: isIntersecting },
    }),
    sentinelRef
  })
```

**Pull request merged**